src/Microsoft.TemplateEngine.Cli/TemplateListCoordinator.cs

```diff
@@ -94,17 +97,29 @@ internal class TemplateListCoordinator
                         GetInputParametersString(args/*, appliedParameterMatches*/))
                     .Bold().Red());
 
-                if (resolutionResult.HasTemplateGroupMatches)
+                if (resolutionResult.HasTemplateGroupMatches && resolutionResult.ListFilterMismatchGroupCount > 0)
```

Comment:
discussion item: 
currently "filters" count includes only the groups that satisfy constraints but failed on filters, and vice versa - the constraints count includes groups which matched filters but failed on constraints.

Thus, if the template group fails on both filters and constraints will not be counted.
Is this behavior ok or should I better change it so both counts include the groups failed on both.

Reply:
This behavior makes sense - since we have different options that the user must specify to fix each case (filter or constraint) it makes sense to have different checks here for each message/condition.